metrics/options.go

```diff
@@ -76,6 +76,13 @@ func WithPrometheusExporterEnabled() Option {
 	}
 }
 
+func WithPrometheusPushgatewayEnabled() Option {
+	return func(opts *Options) {
+		enabled := true
+		opts.Metrics.Prometheus.Pushgateway.Enabled = &enabled
```

Comment:
why use pointer?

Reply:
cause https://github.com/apache/dubbo-go/issues/2335

Reply:
There are multiple PRs in this issue that have resolved this issue

Reply:
Yes,and this pr just add a method to modify the configuration